full-service/src/db/txo.rs

```diff
@@ -2980,6 +2984,155 @@ mod tests {
         );
     }
 
+    // The narrative for this test is that an exchange creates two assigned
+    // subaddresses for their customers, Alice and Bob.
+    // Alice receives 33 MOB from some external entity, followed by 44 MOB.
+    // Bob receives 55 MOB, then 66 MOB from some external entity.
+    // Confirm max spendable for Alice: 77 MOB
+    // Confirm max spendable for Bob: 121 MOB
+    // Confrim max spendable in entire exchange wallet: 198 MOB
+    #[test_with_logger]
+    fn test_list_spendable_for_spend_only_subaddress(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
+
+        let db_test_context = WalletDbTestContext::default();
+        let wallet_db = db_test_context.get_db_instance(logger);
+        let ledger_db = get_test_ledger(5, &[], 12, &mut rng);
+        let mut pooled_conn = wallet_db.get_pooled_conn().unwrap();
+        let conn = pooled_conn.deref_mut();
+
+        let root_id = RootIdentity::from_random(&mut rng);
+        let account_key = AccountKey::from(&root_id);
+        let (account_id_hex, _public_address_b58) = Account::create_from_root_entropy(
+            &root_id.root_entropy,
+            Some(1),
+            None,
+            None,
+            "Exchange Account",
+            "".to_string(),
+            "".to_string(),
+            conn,
+        )
+        .unwrap();
+
+        // Assign an address for Alice
+        let (alice_public_address_b58, alice_subaddress_index) =
+            AssignedSubaddress::create_next_for_account(
+                &account_id_hex.to_string(),
+                "Alice's Subaddress",
+                &ledger_db,
+                &mut wallet_db.get_pooled_conn().unwrap(),
+            )
+            .unwrap();
```

Comment:
Nice, I like that quite a lot - and I just learned about `collect_tuple` so I can assign all in one go 🙌

Reply:
oh [collect_tuple](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.collect_tuple) is interesting, I was hoping there was a more constant way to do it since `["alice", "bob"]` is a compile time size, but collect_tuple should work fairly well in tests with an "unwrap()"